### PR TITLE
wait for async requests to complete

### PR DESCRIPTION
**Pull request by dwickern**

### CODE REVIEW

File: public/docdef.ts [hunk 1]
```diff
@@ -1,41 +1,8 @@
...
-export function getParameterByName(name: string, url?: string) {
-    if (!url) url = window.location.href;
-    name = name.replace(/[\[\]]/g, '\\$&');
-    const regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)');
-    const results = regex.exec(url);
-    if (!results) return null;
-    if (!results[2]) return '';
-    return decodeURIComponent(results[2].replace(/\+/g, ' '));
-}
+import { ngApi } from "./field-formatters";
```
Comment (zachary1006) [inline public/docdef.ts]: The linter is complaining that this needs to be single-quote.

Comment (dwickern) [reply public/docdef.ts]: I don't see any errors when I run `yarn run lint`

**Comment by zachary1006** [comment]
Looks good. Docdefs with no name display correctly, and the rest of the functionality seems to work well. 

We'll need to add the other charts back in to test the other functionality, but I think this can be merged as-is.

File: public/docdef.ts (hunk 1 above)
Comment (zachary1006) [reply public/docdef.ts]: You're right. Not sure why the IDE complains about it.

**Pull request merged**